ioSPI/cryoemio.py

```diff
@@ -0,0 +1,362 @@
+    - hole_diameter                : diameter in nm
+    - hole_thickness_center        : thickness at center in nm
+    - hole_thickness_edge          : thickness at edge in nm.
+
+    *** beam_parameters ***
+    - voltage                      : voltage in kV
+    - energy_spread                : energy spread in V
+    - electron_dose                : dose per image in e/nm**2
+    - electron_dose_std            : standard deviation of dose per image
+
+    *** optics_parameters ***
+    - magnification                : magnification (81000; 105000; 130000)
+    - spherical_aberration         : spherical aberration in mm
+    - chromatic_aberration         : chromatic aberration in mm
+    - aperture_diameter            : diam in um of aperture in back focal plane (50-100)
+    - focal_length                 : focal length in mm of primary lens
+    - aperture_angle               : aperture angle in mrad of the beam furnished by
+                                         the condenser lens
+    - defocus [OPTIONAL]           : nominal defocus value in um
+    - defocus_syst_error           : standard deviation of a systematic error
+                                         added to the nominal defocus, measured
+                                         in um. Same error is added to the defocus
+                                         of every image.
+    - defocus_nonsyst_error        : standard deviation of a nonsystematic error
+                                         added to the nominal defocus and the
+                                         systematic error, measured in um. A new
+                                         value of error is computed for every image.
+    - optics_defocusout [OPTIONAL] : if present, defocus values written to file.
+
+    *** detector_parameters ***
+    - detector_Nx                  : number of pixels on detector along x axis
+    - detector_Ny                  : number of pixels on detector along y axis
+    - detector_pixel_size          : physical pixel size in um
+    - detector_gain                : detector gain: avg number of counts per electron
+    - noise                        : quantized electron waves result in noise
+    - detector_Q_efficiency        : detector quantum efficiency
+    - MTF_params                   : list of 5 MTF parameters
+
+    *** miscellaneous ***
+    - seed [OPTIONAL]              : seed for the run. If not present, random.
+    """
+    parameters = None
+    with open(yaml_file, "r") as f:
+        parameters = yaml.safe_load(f)
+
+    # fill the dictionary
+    dic = {"simulation": {}}
+    if "miscellaneous" in parameters:
+        if "seed" in parameters["miscellaneous"]:
+            dic["simulation"]["seed"] = parameters["miscellaneous"]["seed"]
+        else:
+            random.seed()
+            dic["simulation"]["seed"] = random.randint(0, int(1e10))
+    else:
+        random.seed()
+        dic["simulation"]["seed"] = random.randint(0, int(1e10))
```

Comment:
could check for dic["simulation"]["seed"], and fill if empty, so don't have to have two else statements

Reply:
changed to a try/except loop waiting for a KeyError as a more clear way of accomplishing the same task.